src/main/java/com/google/devtools/build/lib/runtime/commands/BuildCommand.java

```diff
@@ -94,4 +100,31 @@ public BlazeCommandResult exec(CommandEnvironment env, OptionsParsingResult opti
     ExitCode exitCode = new BuildTool(env).processRequest(request, null).getExitCondition();
     return BlazeCommandResult.exitCode(exitCode);
   }
+
+  public static Pair<ExitCode, List<String>> getTargets(CommandEnvironment env, OptionsParsingResult options) {
```

Comment:
Pairs with `null` are probably not the most elegant way of reusing this code between `BuildCommand`/`TestCommand`, suggestions to improve welcome :)

Reply:
Might be cleaner to move to a new class and use a custom exception to signal that whatever was invalid?